src/library/columncache.cpp

```diff
@@ -181,7 +181,7 @@ void ColumnCache::slotSetKeySortOrder(double notation) {
     // A custom COLLATE function was tested, but using CASE ... WHEN was found to be faster
     // see GitHub PR#649
     // https://github.com/mixxxdj/mixxx/pull/649#discussion_r34863809
-    QString keySortSQL("CASE key_id ");
+    QString keySortSQL("CASE %1_id WHEN NULL THEN 0 ");
```

Comment:
I'm afraid it conflicts with below lines because you end up with multiple %1 in the string.